middle_end/flambda2/from_lambda/closure_conversion.ml

```diff
@@ -917,19 +917,35 @@ let close_c_call acc env ~loc ~let_bound_ids_with_kinds
     | false -> prim_desc
     | true ->
       (* [close_c_call0] checks [prim_native_name] to see whether we should
-         invoke the bytecode name or native name. *)
-      let prim_native_name = "" in
-      (* We should override [prim_native_repr_args] and [prim_native_repr_res]
-         so that no special transformations happen to the args and result, as
-         JavaScript functions won't support them.*)
-      let value : Primitive.mode * Lambda.extern_repr =
-        Prim_global, Same_as_ocaml_repr (Base Value)
+         invoke the bytecode name or native name; for JavaScript compilation, we
+         should use the bytecode name unless it involves unboxed products, for
+         the following reason:
+
+         The expected type for JS stubs with unboxed products are different from
+         bytecode C stubs and bytecode-compiled JSOO stubs. Suppose an external
+         takes an unboxed product, e.g. [#(int * int)]. In bytecode, this is
+         passed as a single argument, containing a pointer to a pair; however,
+         in JavaScript, this will be passed as two arguments.
```

Comment:
This line should say "JSIR" not "JavaScript" I think

Reply:
and change "as two arguments." -> "as two arguments, in the same way as for native compilation."